File: python/ingreso.py
import os
import csv
from cama import Cama
import pandas as pd

class Ingreso:
    def __init__(self):
        self.ID = input("Ingrese el documento de la persona: ")
        self.fecha_ingreso = input("Ingrese una fecha de ingreso: (separadas por [-])")  # str
        self.motivo_ingreso = input("Ingrese la enfermedad o el motivo de ingreso")  # str
        
        self.cama_asignada = input("Ingrese la cama que se le va a asignar: (Escriba [AUTOMATICO] para asignarle cualquiera disponible)")  # int
        if self.cama_asignada == "AUTOMATICO":
            TipoCama = input("Ingrese el tipo de cama que desea (Individual o Compartida)")
            self.cama_asignada = Cama.BuscarCamaPorTipo(TipoCama)
        else:
            cama = Cama(self.cama_asignada)

        self.nombre = input("Ingrese el nombre del paciente: ")
        self.edad = input("Ingrese la edad del paciente: ")

    def registrar_ingreso(self):
            """Registrar el ingreso del paciente en el archivo pacientes.csv."""
            PACIENTES = "Archivos/pacientes.csv"
            
            # Estructura del nuevo registro
            nuevo_paciente = {
                "ID": self.ID,
                "Nombre": self.nombre,
                "Edad": self.edad,
                "Diagnóstico": "",  # Vacío por el momento
                "EspecialidadAsignada": "",  # Vacío por el momento
                "CamaAsignada": self.cama_asignada,
                "FechaIngreso": self.fecha_ingreso,
                "HistorialMedico": ""  # Vacío por el momento
            }

            # Verificar si el archivo ya existe y añadir el encabezado si no existe
            archivo_existe = os.path.exists(PACIENTES)
            
            # Guardar los datos en el archivo CSV
            with open(PACIENTES, mode='a', newline='', encoding='utf-8') as file:
                fieldnames = ["ID", "Nombre", "Edad", "Diagnóstico", "EspecialidadAsignada", "CamaAsignada", "FechaIngreso", "HistorialMedico"]
                writer = csv.DictWriter(file, fieldnames=fieldnames)

                # Escribir el encabezado solo si el archivo es nuevo
                if not archivo_existe:
                    writer.writeheader()
                
                # Escribir el nuevo registro de paciente
                writer.writerow(nuevo_paciente)
            
            print(f"Ingreso registrado para el paciente {self.nombre} con ID {self.id_ingreso} en la cama {self.cama_asignada}.")

    # Registrar la fecha de alta del paciente
    def registrar_alta(self, fecha_alta):
        if self.fecha_alta:
            raise ValueError("El paciente ya tiene una fecha de alta registrada.")
        self.fecha_alta = fecha_alta
        self.liberar_cama()
        print(f"Alta registrada para el paciente {self.paciente.nombre} el {self.fecha_alta}")

    # Liberar la cama asignada al paciente
    def liberar_cama(self):
        print(f"La cama {self.cama_asignada} ha sido liberada.")

File: Csharp/HealthTech/Ingreso.cs
namespace HealtTech
{
    using System;
    using System.Collections.Generic;
    using System.IO;
    using System.Linq;

    public class Ingreso
    {
        public string ID { get; set; }
        public string FechaIngreso { get; set; }
        public string MotivoIngreso { get; set; }
        public string CamaAsignada { get; set; }
        public string Nombre { get; set; }
        public string Edad { get; set; }

        public Ingreso()
        {
            // Validación de entrada del ID
            Console.WriteLine("Ingrese el documento de la persona: ");
            ID = Console.ReadLine();
            while (string.IsNullOrEmpty(ID) || !ID.All(char.IsDigit))
            {
                Console.WriteLine("ID inválido. Ingrese el documento de la persona (solo números): ");
                ID = Console.ReadLine();
            }

            // Validación de fecha
            Console.WriteLine("Ingrese una fecha de ingreso (formato yyyy-mm-dd): ");
            FechaIngreso = Console.ReadLine();
            while (!DateTime.TryParse(FechaIngreso, out _))
            {
                Console.WriteLine("Fecha inválida. Ingrese una fecha de ingreso (formato yyyy-mm-dd): ");
                FechaIngreso = Console.ReadLine();
            }

            // Ingreso del motivo
            Console.WriteLine("Ingrese la enfermedad o el motivo de ingreso: ");
            MotivoIngreso = Console.ReadLine();

            // Validación de cama asignada
            Console.WriteLine("Ingrese la cama que se le va a asignar (Escriba [AUTOMATICO] para asignar cualquiera disponible): ");
            CamaAsignada = Console.ReadLine();
            if (CamaAsignada == "AUTOMATICO")
            {
                Console.WriteLine("Ingrese el tipo de cama que desea (Individual o Compartida): ");
                string tipoCama = Console.ReadLine();
                CamaAsignada = Cama.BuscarCamaPorTipo(tipoCama);
            }
            else
            {
                var cama = new Cama(CamaAsignada);
            }

            // Ingreso del nombre
            Console.WriteLine("Ingrese el nombre del paciente: ");
            Nombre = Console.ReadLine();

            // Validación de edad
            Console.WriteLine("Ingrese la edad del paciente: ");
            Edad = Console.ReadLine();
            while (!int.TryParse(Edad, out _))
            {
                Console.WriteLine("Edad inválida. Ingrese la edad del paciente (números solamente): ");
                Edad = Console.ReadLine();
            }
        }

        // Registrar el ingreso del paciente en el archivo pacientes.csv
        public void RegistrarIngreso()
        {
            string rutaPacientes = @"C:\Users\Luis Angel Seoanes O\OneDrive\Escritorio\HealtTech\Csharp\Archivos\pacientes.csv";

            // Estructura del nuevo registro
            var nuevoPaciente = new Dictionary<string, string>
        {
            { "ID", ID },
            { "Nombre", Nombre },
            { "Edad", Edad },
            { "Diagnóstico", "" }, // Vacío por el momento
            { "EspecialidadAsignada", "" }, // Vacío por el momento
            { "CamaAsignada", CamaAsignada },
            { "FechaIngreso", FechaIngreso },
            { "HistorialMedico", "" } // Vacío por el momento
        };

            // Verificar si el archivo ya existe y añadir el encabezado si no existe
            bool archivoExiste = File.Exists(rutaPacientes);

            // Guardar los datos en el archivo CSV
            using (StreamWriter writer = new StreamWriter(rutaPacientes, true))
            {
                var fieldnames = new List<string> { "ID", "Nombre", "Edad", "Diagnóstico", "EspecialidadAsignada", "CamaAsignada", "FechaIngreso", "HistorialMedico" };

                // Escribir el encabezado solo si el archivo es nuevo
                if (!archivoExiste)
                {
                    writer.WriteLine(string.Join(",", fieldnames));
                }

                // Escribir el nuevo registro de paciente
                writer.WriteLine(string.Join(",", nuevoPaciente.Values));
            }

            Console.WriteLine($"Ingreso registrado para el paciente {Nombre} con ID {ID} en la cama {CamaAsignada}.");
        }

        // Registrar la fecha de alta del paciente
        public void RegistrarAlta(string fechaAlta)
        {
            Console.WriteLine($"Alta registrada para el paciente {Nombre} el {fechaAlta}");
            LiberarCama();
        }

        // Liberar la cama asignada al paciente
        public void LiberarCama()
        {
            // Liberar la cama en el archivo camas.csv
            var cama = new Cama(CamaAsignada);
            cama.LiberarCama();
            Console.WriteLine($"La cama {CamaAsignada} ha sido liberada.");
        }
    }


}